pyfolio/plotting.py

```diff
@@ -1660,3 +1662,126 @@ def plot_prob_profit_trade(round_trips, ax=None):
            xlim=(lower_plot, upper_plot), ylim=(0, y.max() + 1.))
 
     return ax
+
+
+def plot_cone(aggregate_returns, bounds, ax,
+              cone_std=(1, 1.5, 2.),
+              color='green',):
+    """
+    Plots the upper and lower bounds of an n standard deviation
+    cone of forecasted cumulative returns.
+
+    Parameters
+    ----------
+    aggregate_returns : pandas.core.frame.DataFrame
+        Cumulative out-of-sample returns.
+    bounds : pandas.core.frame.DataFrame
+        Contains upper and lower cone boundaries. Column names are
+        strings corresponding to the number of standard devations
+        above (positive) or below (negative) the projected mean
+        cumulative returns.
+    ax : matplotlib.Axes, optional
+        Axes upon which to plot.
+    cone_std : int, float, or list of int/float
+        Number of standard devations to use in the boundaries of
+        the cone. If multiple values are passed, cone bounds will
+        be generated for each value.
+    color : str, optional
+        Any matplotlib color
+
+    Returns
+    -------
+    ax : matplotlib.Axes
+        The axes that were plotted on.
+    """
+    for std in cone_std:
+        ax.fill_between(aggregate_returns.index,
+                        bounds[float(std)].iloc[:len(aggregate_returns)],
+                        bounds[float(-std)].iloc[:len(aggregate_returns)],
+                        color=color, alpha=0.5)
+    return ax
+
+
+def plot_multistrike_cones(is_returns, oos_returns, num_samples=1000,
+                           name=None, ax=None, cone_std=(1., 1.5, 2.),
+                           random_seed=None):
+    """
+    Plots the upper and lower bounds of an n standard deviation
+    cone of forecasted cumulative returns. This cone is non-parametric,
+    meaning it does not assume that returns are normally distributed. Redraws
+    a new cone when returns fall outside of last cone drawn
+
+    Parameters
+    ----------
+    is_returns : pandas.core.frame.DataFrame
+        Non-cumulative in-sample returns.
+    oos_returns : pandas.core.frame.DataFrame
+        Non-cumulative out-of-sample returns.
+    num_samples : int
+        Number of samples to draw from the in-sample daily returns.
+        Each sample will be an array with length num_days.
+        A higher number of samples will generate a more accurate
+        bootstrap cone.
+    name : str, optional
+        Plot title
+    ax : matplotlib.Axes, optional
+        Axes upon which to plot.
+    cone_std : int, float, or list of int/float
+        Number of standard devations to use in the boundaries of
+        the cone. If multiple values are passed, cone bounds will
+        be generated for each value.
+    random_seed : int
+        Seed for the pseudorandom number generator used by the pandas
+        sample method.
+
+
+    Returns
+    -------
+    ax : matplotlib.Axes
```

Comment:
This confused me a bit, because it feels like we're returning both `ax` and `fig`, whereas what we're actually doing is returning axes if the caller passed them in, and if not, we're returning a new figure. Could we find a way to make that clearer in the docstring?